mock-esignet-integration-impl/src/main/java/io/mosip/esignet/mock/integration/service/MockKeyBindingWrapperService.java

```diff
@@ -40,50 +41,112 @@
 import io.mosip.kernel.keymanagerservice.service.KeymanagerService;
 import lombok.extern.slf4j.Slf4j;
 import net.minidev.json.JSONObject;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
 
 @ConditionalOnProperty(value = "mosip.esignet.integration.key-binder", havingValue = "MockKeyBindingWrapperService")
 @Component
 @Slf4j
 public class MockKeyBindingWrapperService implements KeyBinder {
 
     public static final String BINDING_SERVICE_APP_ID = "MOCK_BINDING_SERVICE";
+    private static final String OTP_VALUE = "111111";
+    @Value("${mosip.esignet.mock.authenticator.get-identity-url}")
+    private String getIdentityUrl;
 
+    @Value("#{${mosip.esignet.mock.supported.bind-auth-factor-types}}")
+    private List<String> supportedBindAuthFactorTypes;
     @Autowired
     private KeymanagerService keymanagerService;
 
     @Value("${mosip.esignet.binding.key-expire-days}")
     private int expireInDays;
 
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
     private static final Map<String, List<String>> supportedFormats = new HashMap<>();
 
     static {
-        supportedFormats.put("OTP", Arrays.asList("alpha-numeric"));
-        supportedFormats.put("PIN", Arrays.asList("number"));
-        supportedFormats.put("BIO", Arrays.asList("encoded-json"));
-        supportedFormats.put("WLA", Arrays.asList("jwt"));
+        supportedFormats.put("OTP", List.of("alpha-numeric"));
+        supportedFormats.put("PIN", List.of("number"));
+        supportedFormats.put("BIO", List.of("encoded-json"));
+        supportedFormats.put("WLA", List.of("jwt"));
     }
 
 
     @Override
     public SendOtpResult sendBindingOtp(String individualId, List<String> otpChannels,
                                         Map<String, String> requestHeaders) throws SendOtpException {
-        SendOtpResult sendOtpResult = new SendOtpResult(null,"mock", "mock");
-        //TODO
-        return sendOtpResult;
+
+        if (otpChannels == null
+                || !otpChannels.stream().allMatch(this::isSupportedOtpChannel)) {
+            throw new SendOtpException("invalid_otp_channel");
+        }
+
+        try {
```

Comment:
This is a common logic, can we move this to mockHelperService.

Reply:
common logic moved into mockHelperService